sklearn/datasets/_openml.py

```diff
@@ -51,6 +51,8 @@ def wrapper(*args, **kw):
                 return f(*args, **kw)
             except HTTPError:
                 raise
+            except ValueError:
```

Comment:
```py
except (HTTPError, ValueError):
	raise
```

Reply:
Should we also passthrough `ArffException` as well?

Reply:
> Should we also passthrough ArffException as well?

No, because they may be due to data corruption.

Reply:
That was my concern about not letting ValueError through too...

Reply:
There are some `ValueError`s in `_arff.py` as well. In principle, we only want to retry when there is an parsing error or an error from `_open_openml_url`, which is scoped to:

```py
def _load_arff(...):
	response = _open_openml_url(url, data_home)

	with closing(response):
		arff = _arff.load((line.decode('utf-8')
 	    	              for line in response),
    	    	          return_type=return_type,
        	    	      encode_nominal=not as_frame)
	return arff
```

Can we place this in its own function and use the retry wrapper on this new function?

Reply:
No, that code only reads the headers and returns a generator, which is the basis of this change. The parsing errors will actually occur when that generator is iterated during conversion to arrays. 

If the ValueErrors in _arff.py are not possible to raise due to data corruption, then the current solution is fine

Reply:
The only `ValueError` I see from corruption may be https://github.com/scikit-learn/scikit-learn/blob/fa9cf22b2792e63f9ff95cd417723b3359a2c689/sklearn/externals/_arff.py#L243-L249

On a side note, we can define the `_load_arff` as follows:

```py
def _load_arff(..., as_frame):
	response = _open_openml_url(url, data_home)

	with closing(response):
		arff = _arff.load((line.decode('utf-8')
 	    	              for line in response),
    	    	          return_type=return_type,
        	    	      encode_nominal=not as_frame)
		if as_frame:
			return _convert_arff_data_dataframe(arff, columns, features_dict)
		else:
			return _convert_arff_data(arff['data'], col_slice_x,
                              		  col_slice_y, shape)
```